File: pclassic/2016s/PClassic2016Stubs/stubs/MessageOrigins.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.Reader;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
public class MessageOrigins {

	public static void main(String[] args) throws IOException{
		Reader r = new FileReader("MessageOriginsIN.txt");
		BufferedReader reader = new BufferedReader(r);
		while (reader.ready()){
			String[] arr=reader.readLine().split(" ");
			int[] intArr = new int[arr.length - 1];
	
			for(int i=0;i < arr.length - 1;i++)
			 intArr[i]=Integer.parseInt(arr[i]);
			int index = Integer.parseInt(arr[arr.length - 1]);
			System.out.println(findIndex(intArr, index));
		}
		
	}
		
  //Fill out the body of this method
  public static int findIndex(int [] parent, int i){
    return 0;
  }
}

File: pclassic/2016s/PClassic2016Stubs/stubs/MessageOrigins.py
def origin(messages, index):
    return 0

if __name__ == "__main__":
  with open("MessageOriginsIN.txt", "r") as f:
      while True:
          s = f.readline()
          if s == "":
              break
          data = s.split(" ")
          N = int(data[len(data) - 1])
          messages = [int(x) for x in data[:-1]]
          print origin(messages, N)
